docs/source/tutorials/building_blocks.rst

```diff
@@ -0,0 +1,247 @@
+
+            square = Square()
+            square.shift(LEFT)
+
+         can be replaced by
+
+         .. code-block:: python
+
+            square = Square().shift(LEFT)
+
+
+Styling mobjects
+================
+
+The following scene changes the default aesthetics of the mobjects.
+
+.. code-block:: python
+
+   class MobjectStyling(Scene):
+       def construct(self):
+           circle = Circle().shift(LEFT)
+           square = Square().shift(UP)
+           triangle = Triangle().shift(RIGHT)
+
+           circle.set_stroke(color=GREEN, width=20)
+           square.set_fill(YELLOW, opacity=1.0)
+           triangle.set_fill(PINK, opacity=0.5)
+
+           self.add(circle, square, triangle)
+           self.wait(1)
+
+.. image:: ../_static/building_blocks/styling.png
+    :align: center
+    :alt: figure of a circle, a square, and a triangle
+
+This scene uses two of the main functions that change the visual style of a
+mobject: :meth:`.set_stroke` and :meth:`.set_fill`.  The former changes the
+visual style of the mobject's border while the latter changes the style of the
+interior.  By default, most mobjects have a fully transparent interior so you
+must specify the ``opacity`` parameter in order to display the color.  An
+opacity of ``1.0`` means fully opaque, while ``0.0`` means fully transparent.
+
+Only instances of :class:`.VMobject` implement :meth:`.set_stroke` and
+:meth:`.set_fill`.  Instances of :class:`.Mobject` implement
+:meth:`.~Mobject.set_color` instead.  The vast majority of pre-defined classes
+are derived from :class:`.VMobject` so it is usually safe to assume that you
+have access to :meth:`.set_stroke` and :meth:`.set_fill`.
+
+
+Mobject on-screen order
+=======================
+
+The next scene is exactly the same as the ``MobjectStyling`` scene from the
+previous section, except for exactly one line.
+
+.. code-block:: python
+
+   class MobjectZOrder(Scene):
+       def construct(self):
+           circle = Circle().shift(LEFT)
+           square = Square().shift(UP)
+           triangle = Triangle().shift(RIGHT)
+
+           circle.set_stroke(color=GREEN, width=20)
+           square.set_fill(YELLOW, opacity=1.0)
+           triangle.set_fill(PINK, opacity=0.5)
+
+           self.add(triangle, square, circle)
+           self.wait(1)
+
+The only difference here (besides the scene name) is the order in which the
+mobjects are added to the scene.  In ``MobjectStyling``, we added them as
+``add(circle, square, triangle)``, whereas in ``MobjectZOrder`` we add them as
+``add(triangle, square, circle)``.  The result is the following:
+
+.. image:: ../_static/building_blocks/zorder.png
+    :align: center
+    :alt: figure of a circle, a square, and a triangle
+
+As you can see, the order of the arguments of :meth:`~.Scene.add` determines
+the order that the mobjects are displayed on screen, with the left-most
+arguments being put in the back.
+
+.. tip:: The display order of mobjects is called the *z-index*.
```

Comment:
**Suggestion (if minor)** : Maybe add the way to set z-index (`set_z_index`).

Reply:
I tried, but the current implementation of `z_index` does not play well with `add` and `remove`. So the tutorial I _wanted_ to write is actually not possible at this moment.